quipucords/scanner/openshift/api.py

```diff
@@ -262,8 +262,14 @@ def retrieve_operators(self, **kwargs) -> List[ClusterOperator | LifecycleOperat
             for csv in self._list_cluster_service_versions(**kwargs)
         }
         for operator in olm_operators:
-            csv = csv_map[operator.cluster_service_version]
-            operator.display_name = csv.spec.displayName
+            try:
+                csv = csv_map[operator.cluster_service_version]
+                operator.display_name = csv.spec.displayName
+            except KeyError:
+                # This can happen in rare cases when we find an operator that wasn't
+                # listed in the cluster service versions. I'm not sure how this can
+                # happen, but we have actually seen it on a shared OpenShift server.
+                operator.display_name = operator.name
```

Comment:
considering `display_name` is optional in its pydantic model, I'd leave it empty here. Thoughts?

Reply:
That's true! I'll leave it not set and generate another report to confirm that it's still legible. If it seems good, I may just replace this with a `pass`.

Reply:
also, the intention of `display_name` was to show an user friendly name as displayed on catalog. I'd rather have it set as null and hence indicating discovery failed to find it instead of duplicating the non user friendly name from olm

Reply:
I have a personal preference when it comes to `except SomeExc: pass`: use [contextlib suppress](https://docs.python.org/3/library/contextlib.html#contextlib.suppress)

Feel free to do whatever u prefer, tho.

Reply:
I haven't used `contextlib.suppress` before, but it looks like a good fit here. Thanks for the suggestion!